frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -295,6 +301,17 @@ fn construct_runtime_final_expansion(
 		impl #scrate::sp_runtime::traits::GetRuntimeBlockType for #name {
 			type RuntimeBlock = #block;
 		}
+		// The outer enums are generated in this scope with the following names.
+		impl #scrate::sp_runtime::traits::GetRuntimeOuterEnumTypes for #name {
+			// The `RuntimeCall` must be defined for all runtimes.
+			type RuntimeCall = <#name as #system_path::Config>::RuntimeCall;
+			// The associated type `RuntimeEvent` is present in the `frame_system::Config` trait
+			// if and only if the `#[pallet::event]` is declared. However, fallback to our
+			// generated outer `RuntimeEvent` enum to ensure consistency with pallets that don't
+			// emit events.
+			type RuntimeEvent = RuntimeEvent;
+			type RuntimeError = RuntimeError;
+		}
```

Comment:
I don't get the need for this trait.

Reply:
We could probably get the path from the `runtime` variable in the metadata collection, then presume we have the `RuntimeCall`, `RuntimeEvent`, and `RuntimeError` enums present. 

I've added this assuming that users could override the `RuntimeCall` via `frame_system::Config`, let me know if I've got this right 🙏

Reply:
I mean the `construct_runtime` macro is already generating all the types that you are needing when generating the metadata.

The argument about overriding `RuntimeCall` is probably very niche, but fine. Still, you don't need the `GetRuntimeOuterEnumTypes` trait.

The comment above `RuntimeEvent` is also not correct, as the type in the `Config` is always there.

Reply:
Have removed this marker trait and used the information directly, thanks!